gensim/models/sent2vec.py

```diff
@@ -0,0 +1,307 @@
+            sys.exit()
+
+    def threshold(self, t):
+        self.words = [entry for entry in self.words if entry.count > t]
+        self.size = 0
+        self.word2int = [-1] * self.max_vocab_size
+        for entry in self.words:
+            h = self.find(entry.word)
+            self.word2int[h] = self.size
+            self.size += 1
+
+    def initTableDiscard(self):
+        for i in range(self.size):
+            f = self.words[i].count / self.ntokens
+            self.pdiscard.append(((self.t / f)**(0.5)) + (self.t / f))
+
+    def initNgrams(self):
+        for i in range(self.size):
+            self.words[i].subwords.append(i)
+            word = self.words[i].word
+            for j in range(len(word)):
+                ngram = ""
+                for k, n in zip(range(j, len(word)), range(1, self.maxn + 1)):
+                    ngram += word[k]
+                    k += 1
+                    while k < len(word):
+                        ngram += word[k]
+                        k += 1
+                    if n >= self.minn and ((n == 1 and (j == 0 or k == len(word))) is False):
+                        h = self.hash_(ngram) % self.bucket
+                        self.words[i].subwords.append(self.size + h)
+
+    def add_ngrams_train(self, context, n, k):
+        line = list(context)
+        num_discarded = 0
+        line_size = len(line)
+        discard = [False] * line_size
+        while num_discarded < k and line_size - num_discarded > 2:
+            token_to_discard = randint(0, line_size - 1)
+            if discard[token_to_discard] is False:
+                discard[token_to_discard] = True
+                num_discarded += 1
+        for i in range(line_size):
+            if discard[i] is True:
+                continue
+            h = line[i]
+            for j in range(i + 1, line_size):
+                if j >= i + n or discard[j] is True:
+                    break
+                h = h * 116049371 + line[j]
+                line.append(self.size + (h % self.bucket))
+        return line
+
+    def add_ngrams(self, context, n):
+        line = list(context)
+        line_size = len(context)
+        for i in range(line_size):
+            h = line[i]
+            for j in range(i + 1, line_size):
+                if j >= i + n:
+                    break
+                h = h * 116049371 + line[j]
+                line.append(self.size + (h % self.bucket))
+        return line
+
+    def get_line(self, sentence):
+        hashes = []
+        words = []
+        ntokens = 0
+        for word in sentence:
+            h = self.find(word)
+            wid = self.word2int[h]
+            if wid < 0:
+                hashes.append(self.hash_(word))
+                continue
+            ntokens += 1
+            words.append(wid)
+            hashes.append(self.hash_(word))
+            if ntokens > self.max_line_size:
+                break
+        return ntokens, hashes, words
+
+
+class Model():
```

Comment:
Are you really need this class (why this isn't a part of `Sent2Vec`?)

Reply:
Done. Merged with sent2vec now.